src/Symfony/Component/Amqp/Broker.php

```diff
@@ -0,0 +1,481 @@
+     *
+     * If it's not the case, you MUST specify the queueName.
+     */
+    public function retry(\AMQPEnvelope $msg, string $queueName = null, string $retryMessage = null): bool
+    {
+        $queueName = $queueName ?: $msg->getRoutingKey();
+
+        $queue = $this->getQueue($queueName);
+
+        if (!$queue->getRetryStrategy()) {
+            throw new LogicException(sprintf('The queue "%s" has no retry strategy.', $queueName));
+        }
+
+        $retryStrategy = $queue->getRetryStrategy();
+
+        if (!$retryStrategy->isRetryable($msg)) {
+            throw new NonRetryableException($retryStrategy, $msg);
+        }
+
+        $time = $retryStrategy->getWaitingTime($msg);
+
+        $this->createDelayedQueue($queueName, $time);
+
+        // Copy previous headers, but omit x-death
+        $headers = $msg->getHeaders();
+        unset($headers['x-death']);
+        $headers['queue-time'] = (string) $time;
+        $headers['exchange'] = (string) self::RETRY_EXCHANGE;
+        $headers['retries'] = $msg->getHeader('retries') + 1;
+
+        // Some RabbitMQ versions fail when $retryMessage is null
+        if (null !== $retryMessage) {
+            $headers['retry-message'] = $retryMessage;
+        }
+
+        return $this->publish($queueName, $msg->getBody(), array(
+            'exchange' => self::DEAD_LETTER_EXCHANGE,
+            'headers' => $headers,
+        ));
+    }
+
+    /**
+     * Moves a message to a given route.
+     *
+     * If attributes are given as third argument they will override the
+     * message ones.
+     */
+    public function move(\AMQPEnvelope $msg, string $routingKey, array $attributes = array()): bool
+    {
+        return $this->simpleBroker->move($msg, $routingKey, $attributes);
+    }
+
+    public function moveToDeadLetter(\AMQPEnvelope $msg, array $attributes = array()): bool
+    {
+        return $this->simpleBroker->moveToDeadLetter($msg, $attributes);
+    }
+
+    /**
+     * Sends a message with delay.
+     *
+     * The message is stored in a pending queue before it's in the expected
+     * target.
+     *
+     * If the target queue is not created, it will be created with default
+     * configuration.
+     */
+    public function delay(string $routingKey, string $message, int $delay, array $attributes = array()): bool
+    {
+        $exchangeName = $attributes['exchange'] ?? self::DEFAULT_EXCHANGE;
+
+        $this->createDelayedQueue($routingKey, $delay, $exchangeName);
+
+        $attributes['exchange'] = self::DEAD_LETTER_EXCHANGE;
+        $attributes['headers']['queue-time'] = (string) $delay;
+        $attributes['headers']['exchange'] = (string) $exchangeName;
+
+        return $this->publish($routingKey, $message, $attributes);
+    }
+
+    public function consume(string $queueName, callable $callback = null, int $flags = \AMQP_NOPARAM, string $consumerTag = null)
+    {
+        $this->getQueue($queueName);
```

Comment:
Looks like `getQueue` does more than getting the queue. I'd explicit this in the name of the method.